harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Aligner.java

```diff
@@ -232,21 +223,44 @@ public HarvestResult align(Set<RecordInfo> records, List<HarvestError> errors) t
 
                 }
 
+                dataMan.forceIndexChanges();
                 result.totalMetadata++;
             } catch (Throwable t) {
                 errors.add(new HarvestError(this.context, t));
                 log.error("Unable to process record from csw (" + this.params.getName() + ")");
                 log.error("   Record failed: " +  ri.uuid + ". Error is: " + t.getMessage());
+				log.error(t);
             } finally {
                 result.originalMetadata++;
             }
         }
-        dataMan.forceIndexChanges();
+	}
 
-        log.debug("End of alignment for : " + params.getName());
+    /**
+     * Remove records no longer on the remote CSW server
+     * 
+     * @param records
+     * @throws Exception
+     */
+	@Transactional(value=TxType.REQUIRES_NEW)
```

Comment:
With this change a transaction is created for every update?

Reply:
Yes. It is not yet the perfect solution, but before that we were doing it anyway with the forcibly flush(), which is much more disrupting than telling spring to just use a new transaction and handle it normally.